flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/filesystem/FileSystemTableSinkTest.java

```diff
@@ -0,0 +1,214 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.filesystem;
+
+import org.apache.flink.api.common.JobExecutionResult;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.dag.Transformation;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSink;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.graph.StreamGraph;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.connector.sink.DataStreamSinkProvider;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.util.ExceptionUtils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+/** Test for {@link FileSystemTableSink}. */
+public class FileSystemTableSinkTest {
+
+    private static final TableSchema TEST_SCHEMA =
+            TableSchema.builder()
+                    .field("f0", DataTypes.STRING())
+                    .field("f1", DataTypes.BIGINT())
+                    .field("f2", DataTypes.BIGINT())
+                    .build();
+
+    @Test
+    public void testFileSytemTableSinkWithParaallelismInChangeLogMode() {
+
+        int parallelism = 2;
+
+        DescriptorProperties descriptor = new DescriptorProperties();
+        descriptor.putString(FactoryUtil.CONNECTOR.key(), "filesystem");
+        descriptor.putString("path", "/tmp");
+        descriptor.putString("format", "testcsv");
+        descriptor.putString(
+                TestCsvFileSystemFormatFactory.IDENTIFIER
+                        + "."
+                        + TestCsvFileSystemFormatFactory.USE_UPSERT_CHANGELOG_MODE.key(),
+                "true");
+        descriptor.putString(FactoryUtil.SINK_PARALLELISM.key(), String.valueOf(parallelism));
+
+        final DynamicTableSink tableSink = createSink(descriptor);
+        Assert.assertTrue(tableSink instanceof FileSystemTableSink);
+
+        final DynamicTableSink.SinkRuntimeProvider provider =
+                tableSink.getSinkRuntimeProvider(new MockSinkContext(false));
+        Assert.assertTrue(provider instanceof DataStreamSinkProvider);
+
+        final DataStreamSinkProvider dataStreamSinkProvider = (DataStreamSinkProvider) provider;
+        try {
+            dataStreamSinkProvider.consumeDataStream(createInputDataStream());
```

Comment:
We should add `fail(...)` to make sure it will fail if no exception is thrown. 

This can be simplified into 

```java
import static org.apache.flink.core.testutils.CommonTestUtils.assertThrows;
assertThrows(
                "xxxx",
                ValidationException.class,
                () -> dataStreamSinkProvider.consumeDataStream(createInputDataStream()));
```